File: Algorithms/11. Container With Most Water/golang/solution.go
func maxArea(height []int) int {
	left, right, area := 0, len(height)-1, 0
	var vol int
	for left < right {
		if height[right] > height[left] {
			vol = (right - left) * height[left]
			left++
		} else {
			vol = (right - left) * height[right]
			right--
		}
		if vol > area {
			area = vol
		}
	}
	return area
}

File: Algorithms/11. Container With Most Water/php/solution.php
<?php
class Solution
{

    /**
     * @param Integer[] $height
     * @return Integer
     */
    function maxArea($height)
    {
        $area = 0;
        $left = 0;
        $right = count($height) - 1;

        while ($left < $right) {
            $area = max($area, min($height[$left], $height[$right]) * ($right - $left));
            if ($height[$left] < $height[$right]) {
                $left++;
            } else {
                $right--;
            }
        }
        return $area;
    }
}
